tools/ARIAtools/extractProduct.py

```diff
@@ -474,6 +511,48 @@ def merged_productbbox(
     else:
         bbox_file = prods_TOTbbox
 
+    # Compare current bbox to existing bbox
+    if exist_bbox and run_log:
+        exist_area = ARIAtools.util.shp.shp_area(exist_bbox, lyr_proj)
+
+        # Calculate overlap area
+        new_bbox = ARIAtools.util.shp.open_shp(prods_TOTbbox)
+        new_area = ARIAtools.util.shp.shp_area(new_bbox, lyr_proj)
+        area_ratio = new_area / exist_area
+
+        olap_bbox = exist_bbox.intersection(new_bbox)
+        olap_area = ARIAtools.util.shp.shp_area(olap_bbox, lyr_proj)
+
+        # Compare areas
+        delta_area = np.abs(olap_area - exist_area)
+        delta_area = np.round(delta_area * 1E7) * 1E-7
+
+        olap_ratio = olap_area / exist_area
+        olap_ratio = np.round(olap_ratio * 1E7) * 1E-7
+
+        if verbose:
+            print(f'Area difference (|prev - new|): {delta_area:.7f} km\u00b2')
+            print(f'Area ratio (new/prev): {area_ratio:.7f}')
+            print(f'Overlap ratio (new/prev): {olap_ratio:.7f}')
+
+        if (delta_area != 0.0) or (olap_ratio != 1.0):
+            LOGGER.debug(f'Product bbox changed in size from previous run '
+                         f'{new_area} vs {exist_area}')
+
+        if shapely.equals(new_bbox, exist_bbox):
+            # Same bbox within machine precision
+            update_mode = 'skip'
+        elif olap_ratio < 1.0:
```

Comment:
would be good to add loggin.WARN if olap_ration falls below some threshold, e.g. 0.6 (60% change0 to warn about significantly smaller area